spring-cloud-dataflow-docs/src/main/asciidoc/streams.adoc

```diff
@@ -135,6 +145,62 @@ If directed graphs are needed instead of the simple linear streams described abo
 == Module Labels
 
 When a stream is comprised of multiple modules with the same name, they must be qualified with labels:
-
+```
   stream create --definition "http | firstLabel: transform --expression=payload.toUpperCase() | secondLabel: transform --expression=payload+'!' | log" --name myStreamWithLabels --deploy
+```
+[[tap-dsl]]
+== Tap DSL
+
+Taps can be created at various producer endpoints in a stream. For a stream like this:
+
+```
+  stream create --definition "http | step1: transform --expression=payload.toUpperCase() | step2: transform --expression=payload+'!' | log" --name mainstream --deploy
+
+```
+taps can be created at the output of `http`, `step1` and `step2`.
+
+To create a stream that acts as a 'tap' on another stream requires to specify the `source destination name` for the tap stream. The syntax for source destination name is:
+
+```
+`<stream-name>.<label/app name>`
+```
+To create a tap at the output of `http` in the stream above, the source destination name is `mainstream.http`
+To create a tap at the output of first transform app in the stream above, the source destination name is `mainstream.step1`
+
+The tap stream DSL looks like this:
+
+```
+  stream create --definition ":mainstream.http > counter" --name tap_at_http --deploy
+
+  stream create --definition ":mainstream.step1 > jdbc" --name tap_at_step1_transformer --deploy
+
+```
+
+Note the colon (:) prefix before the destination names. The colon lets the parser parses this as the destination name instead of app name.
+
+[[explicit-destination-names]]
+== Connecting to explicit destination names at the broker
+
+One can connect to a specific source destination name located in the broker (Rabbit, Kafka) and connect it to a downstream set of apps.
+In this case, the DSL looks like this:
+
+```
+  stream create --definition ":mySource > log" --name ingest_from_broker --deploy
```

Comment:
I think "mySource" is a confusing name here since it might imply a correlation to a `source` app in the `spring-cloud-stream` sense. Maybe just call it "myDestination" or something.